R/Rules-methods.R

```diff
@@ -773,6 +773,59 @@ setMethod(
     }
 )
 
+
+# maxDose-IncrementsHSRBeta ----
+
+#' @rdname maxDose
+#'
+#' @description Determine the maximum possible dose for escalation.
+#'
+#' @aliases maxDose-IncrementsHSRBeta
+#' @example examples/Rules-method-maxDose-IncrementsHSRBeta.R
+#' @export
+setMethod(
+  "maxDose",
+  signature = signature(
+    increments = "IncrementsHSRBeta",
+    data = "Data"
+  ),
+  definition = function(increments, data, ...) {
+    # summary of observed data per dose level
+    y <- factor(data@y, levels = c("0", "1"))
+    dlt_tab <- table(y, data@x)
+
+    # ignore placebo if applied
+    if (data@placebo == TRUE & min(data@x) == data@doseGrid[1]) {
```

Comment:
what if there is placebo and the first dose grid is not equal to min of x?

Reply:
The table with DLTs is created based on the actual data. If placebo is not given, there is no column for placebo and it needs not to be removed. I think this is ok I added a test case to check if the methods works as expected.